adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -266,6 +279,41 @@ func (apiReconciler *APIReconciler) getAuthenticationsForResources(ctx context.C
 	return authentications, nil
 }
 
+func (apiReconciler *APIReconciler) getBackendProperties(ctx context.Context,
+	httpRoute *gwapiv1b1.HTTPRoute) dpv1alpha1.BackendPropertyMapping {
+	backendPropertyMapping := make(dpv1alpha1.BackendPropertyMapping)
+	for ruleIndex, rule := range httpRoute.Spec.Rules {
```

Comment:
why don't you just use the index to get all services attached to the httproute?

Reply:
If we use the index it will give a list of services but cannot say which rule the service is attached to when there are multiple rules in a httproute

Reply:
why not follow the same way we have done for authentications?

Reply:
I checked the code for authentications, It uses a `httprouteToAuthentication` index created using targetRef(which is a httpRoute again) property of Authentication resources. So it is possible to list all authentications for a httpRoute using this index. 

In here. We have created a `serviceToHttpRoute` index using backendRef(service) section of HTTPRoute resources. Since the index is not a `httpRouteToService` index it is not possible to give a httpRoute and list all services. if it were then can give a httpRoute to list all the services. If we had a ref to httproute from Service resource (same as there in the Authentication) then only it is possible to create an index like that.

So `serviceToHttpRoute` index is only utilised for reconcile event filtering when a service changes